cmd/flat-network-controller/app/dns_deployment_test.go

```diff
@@ -0,0 +1,151 @@
+					Name:   "net-sa-rbac-exist",
+					Tenant: "test-te",
+				},
+				Spec: v1.NetworkSpec{
+					Type: "flat",
+				},
+				Status: v1.NetworkStatus{},
+			},
+			expectedTestTag: tagValObjExist,
+			objects: []runtime.Object{
+				&corev1.ServiceAccount{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "coredns",
+						Namespace:   "kube-system",
+						Tenant:      "test-te",
+						Annotations: map[string]string{testTagKey: tagValObjExist},
+					},
+				},
+				&rbacv1.ClusterRole{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "system:coredns",
+						Tenant:      "test-te",
+						Annotations: map[string]string{testTagKey: tagValObjExist},
+					},
+				},
+				&rbacv1.ClusterRoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "system:coredns",
+						Tenant:      "test-te",
+						Annotations: map[string]string{testTagKey: tagValObjExist},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			namePerNetwork := "coredns-" + tc.input.Name
+			kubeClient := fake.NewSimpleClientset(tc.objects...)
+			err := deployDNSForNetwork(tc.input, kubeClient, "cluster.local")
+
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			sa, err := kubeClient.CoreV1().ServiceAccountsWithMultiTenancy("kube-system", tc.input.Tenant).Get("coredns", metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("failed to find sa %s/kube-syste/coredns: %v", tc.input.Tenant, err)
+			}
+			if len(tc.expectedTestTag) > 0 {
+				val := sa.Annotations[testTagKey]
+				if val != tc.expectedTestTag {
+					t.Errorf("expected the existent object; got %v", sa)
+				}
+			}
+
+			clusterRole, err := kubeClient.RbacV1().ClusterRolesWithMultiTenancy(tc.input.Tenant).Get("system:coredns", metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("failed to find cluster role %s/system:coredns: %v", tc.input.Tenant, err)
+			}
+			if len(tc.expectedTestTag) > 0 {
+				val := clusterRole.Annotations[testTagKey]
+				if val != tc.expectedTestTag {
+					t.Errorf("expected the existent object; got %v", sa)
+				}
+			}
+
+			clusterRoleBinding, err := kubeClient.RbacV1().ClusterRoleBindingsWithMultiTenancy(tc.input.Tenant).Get("system:coredns", metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("failed to find cluster role %s/system:coredns: %v", tc.input.Tenant, err)
+			}
+			if len(tc.expectedTestTag) > 0 {
+				val := clusterRoleBinding.Annotations[testTagKey]
+				if val != tc.expectedTestTag {
+					t.Errorf("expected the existent object; got %v", sa)
+				}
+			}
+
+			_, err = kubeClient.CoreV1().ConfigMapsWithMultiTenancy("kube-system", tc.input.Tenant).Get(namePerNetwork, metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("failed to find configmap %s/kube-system/%s: %v", tc.input.Tenant, namePerNetwork, err)
+			}
+
+			_, err = kubeClient.AppsV1().DeploymentsWithMultiTenancy("kube-system", tc.input.Tenant).Get(namePerNetwork, metav1.GetOptions{})
```

Comment:
Nit: kube-syste -> metav1.NamepsaceSystem

Reply:
test code is fine to have literals.